engine/src/main/java/org/terasology/engine/rendering/nui/layers/mainMenu/advancedGameSetupScreen/AdvancedGameSetupScreen.java

```diff
@@ -779,7 +783,7 @@ private void refreshSelection() {
         updateValidToSelect();
     }
 
-    public void setUniverseWrapper(UniverseWrapper wrapper) {
-        universeWrapper = wrapper;
+    public void setEnvironment(UniverseWrapper wrapper) {
+        CoreRegistry.put(UniverseWrapper.class, wrapper);
```

Comment:
Theoretically, the idea was to do the following:
```
context.put(UniverseWrapper.class, wrapper);
CoreRegistry.setContext(context);
```
However, this does not work and leads to an NPE in `UniverseSetupScreen.java:182` when attempting to access `UniverseWrapper` from a context that it's not in. At this moment, it's unclear, why this does not work.

Reply:
Do you want to persist this information in the code as comment instead of just here on this PR?

Reply:
done